src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -353,6 +353,8 @@ public void GenerateInlineSummary(RawComment comment)
 
         public void GenerateClass(Class @class)
         {
+            @class.IsStruct = ExtensionMethods.CheckClassIsStructible(@class, Driver);
```

Comment:
We should not be modifying the AST in the text generators. This should be done in a pass.
 Add a line note